ibid/source/slack.py

```diff
@@ -0,0 +1,371 @@
+            self.handleError(data)
+
+        # Ignore other bots
+        elif data['type'] == 'message' and data.get('subtype') != 'bot_message':
+
+            # Handle annoying diff versions of Slack servers
+            if 'message' in data:
+                sender = self.factory.users[data['message']['user']]['name']
+                text = data['message']['text']
+                is_edited = 'edited' in data['message']
+            else:
+                sender = self.factory.users[data['user']]['name']
+                text = data['text']
+                is_edited = 'edited' in data
+
+            # Ignore edits to sent messages
+            if is_edited:
+                log.info("Ignoring edited message")
+                return
+
+            channel_id = data['channel']
+            raw_text = text
+            event = Event(u'slack', u'message')
+            event.message = self.factory.clean_slack_msg(raw_text)
+            event.channel = channel_id
+            event.sender['id'] = sender
+            event.sender['connection'] = channel_id
+            event.sender['nick'] = sender
+            log.debug(u'Got raw message: {}'.format(event.message))
+            if channel_id[0] == 'D':
+                event.public = False
+                event.addressed = True
+            else:
+                event.public = True
+                if re.search(r'<@{}>'.format(self.factory.user_id), raw_text):
+                    event.addressed = True
+            ibid.dispatcher.dispatch(event).addCallback(self.parseResponse)
+
+    def onClose(self, wasClean, code, reason):
+        log.error(u'WebSocket connection closed: {0}'.format(reason))
+
+    def parseResponse(self, event):
+        self.messageNumber += 1
+        log.debug(u'Responding to event via wss://')
+        for response in event.responses:
+            data = {
+                u'id': self.messageNumber,
+                u'type': u'message',
+                u'channel': response['target'],
+                u'text': response['reply'].encode('utf-8')
+            }
+            self.sendMessage(json.dumps(data), isBinary=False)
+
+    def respondHTTP(self, event):
+        log.debug(u'Responding to event via http://')
+        for response in event.responses:
+            self.factory.post_message(event, response)
+
+
+class SlackBot(WebSocketClientFactory):
+
+    protocol = WebSocketSlackProtocol
+
+    def __init__(self, team, token, channels=None, groups=None):
+        self.team = team
+        self.token = token
+        self.socket_url = ''
+        self.authenticated = False
+        self.channels_to_join = channels
+        self.groups_to_join = groups
+        self.channels = []
+        self.groups = []
+        self.users = {}
+        self.user_id = None
+        self.name = ''
+
+    def clean_slack_msg(self, raw_msg):
+        """ For no good reason slack uses <@`uid`> instead of
+        `name`: so we have to clean it up a little
+        """
```

Comment:
sounds like a good idea -- do other platforms also need to clean the raw messages?